wai-extra/Network/Wai/Middleware/RequestSizeLimit.hs

```diff
@@ -0,0 +1,40 @@
+module Network.Wai.Middleware.RequestSizeLimit (requestSizeLimitMiddleware) where
+
+import Network.Wai
+import Network.Wai.Request
+import qualified Data.ByteString.Lazy as BSL
+import Data.Word (Word64)
+import Network.HTTP.Types.Status (requestEntityTooLarge413)
+import qualified Data.ByteString.Lazy.Char8 as LS8
+import Control.Exception (try, catch)
+
+-- | Middleware to limit request bodies to a certain size. 
+-- If the body is too large, this middleware returns HTTP 413, with a plain text body describing the error.
+--
+-- This uses 'requestSizeCheck' under the hood; see that function for details.
+--
+-- @since 3.1.1
+requestSizeLimitMiddleware :: Word64 -> Middleware
```

Comment:
Thoughts on if this should take some optional function to allow a custom error? The main use case I can think of is if someone wanted to e.g. do a JSON or XML response, or a different status code for some reason.

Reply:
Yeah, it probably should. The most future-proofed way of doing this would be with the builder pattern: have a `RequestSizeLimitSettings` type, a `mkRequestSizeLimitSettings :: Word64 -> RequestSizeLimitSettings`, `requestSizeLimitMiddleware :: RequestSizeLimitSettings -> Middleware`, and then functions like `setErrorResponse :: Response -> RequestSizeLimitSettings -> RequestSizeLimitSettings`.